### DISCUSSION TITLE
compile error on android

### DISCUSSION THREAD

**Discussion by MrLee73** [adrianosantostreina/MobilePermissions#25] [General]
function TMobilePermissionsManifest.ACCESS_CHECKIN_PROPERTIES: JavaString;
begin
  Result := EmptyStr;
end;
and more other like that 

[DCC Error] MobilePermissions.JNI.Os.Base.pas(173): E2010 Incompatible types: 'JString' and 'string'

I try to fix it with replace EmptyStr to **JavaString(EmptyStr)**

and other error come again laike this

function TUsesPermissionsFMXStandard.GetCallCompanionApp: Boolean;
begin
  Result := HasPermission(JavaClass.CALL_COMPANION_APP);
end;

[DCC Error] MobilePermissions.UsesPermissions.FMX.Standard.pas(163): E2003 Undeclared identifier: 'CALL_COMPANION_APP'

**Comment by adrianosantostreina** [comment]
Hello, 

What version of Delphi are you using? 
Are you using the newest version of the component on GitHub? 
I'm using it normally here, no bugs. 

Adriano Santos 


De: "MrLee73" ***@***.***> 
Para: "adrianosantostreina" ***@***.***> 
Cc: "Subscribed" ***@***.***> 
Enviadas: Terça-feira, 15 de novembro de 2022 23:56:51 
Assunto: [adrianosantostreina/MobilePermissions] compile error on android (Discussion #25) 





function TMobilePermissionsManifest.ACCESS_CHECKIN_PROPERTIES: JavaString; 
begin 
Result := EmptyStr; 
end; 
and more other like that 

[DCC Error] MobilePermissions.JNI.Os.Base.pas(173): E2010 Incompatible types: 'JString' and 'string' 

— 
Reply to this email directly, [ https://github.com/adrianosantostreina/MobilePermissions/discussions/25 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/AFSBSUIC57B56QNJORDFKLDWIREPHANCNFSM6AAAAAASBWDZBY | unsubscribe ] . 
You are receiving this because you are subscribed to this thread. Message ID: <adrianosantostreina/MobilePermissions/repo-discussions/25 @ github . com> 


-- 


#MaiorCanalDelphiMobile 

Adriano Santos 
Tornando você um profissional mais completo 

Site oficial: [ http://adrianosantostreina.com.br/ | http://adrianosantostreina.com.br ] 
YouTube: [ http://youtube.com/adrianosantostreina | http://youtube.com/adrianosantostreina ] 
Instagram: [ http://instagram/adrianosantostreina | http://instagram/adrianosantostreina ]

**Reply by MrLee73** [reply]
yes i used newest version of the component on GitHub with 
Delphi 104.2 and Delphi 11.2 with latest update,
both are same error when i compile/run the application for android,

unit Androidapi.JNI.Os does not have the JavaClass.CALL_COMPANION_APP property


it's not happen when I used mobilepermission 0.5.

https://github.com/adrianosantostreina/MobilePermissions/archive/refs/tags/0.5.zip.

The version 0.5 file is running well

**Comment by u253195** [comment]
I had the same erros.

**Comment by fabioaureliosp** [comment]
I have the same problem.

I'm using the updated version of the component, with Delphi 11.2 Update 2.

Thanks

**Comment by romuloTolfo** [comment]
obtive o mesmo problema, sem solução até o momento.

**Comment by adrianosantostreina** [comment]
Romulo, o componente é de código aberto, OpenSource.
Construímos ele de forma voluntária e totalmente gratuito. Infelizmente ainda não conseguimos concluir as modificações.

Entretanto, por ser OpenSource, qualquer membro da comunidade pode fazer um Fork do projeto, corrigir e enviar o PullRequest para que possamos receber as modificações, aprovar e disponibilizar para a comunidade.

E será muito bem vinda a contribuição. 

Não tem interesse em fazer isso?
